.circleci/create_circleci_config.py

```diff
@@ -98,11 +102,52 @@ def to_dict(self):
         pytest_flags.append(
             f"--make-reports={self.name}" if "examples" in self.name else f"--make-reports=tests_{self.name}"
         )
+
+        # We need explicit list instead of `pipeline.parameters.tests_to_run` (only available at job runtime)
+        tests = self.tests_to_run
+        if tests is None:
+            folder = os.environ["test_preparation_dir"]
+            test_file = os.path.join(folder, "filtered_test_list.txt")
+            if os.path.exists(test_file):
+                with open(test_file) as f:
+                    tests = f.read().split(" ")
+
+        # expand the test list
+        if tests == ["tests"]:
+            tests = [os.path.join("tests", x) for x in os.listdir("tests")]
+        expanded_tests = []
+        for test in tests:
+            if test.endswith(".py"):
+                expanded_tests.append(test)
+            elif test == "tests/models":
+                expanded_tests.extend([os.path.join(test, x) for x in os.listdir(test)])
+            elif test == "tests/pipelines":
+                expanded_tests.extend([os.path.join(test, x) for x in os.listdir(test)])
+            else:
+                expanded_tests.append(test)
```

Comment:
Might be easier and more resiliant to just keep iterating as long as you have a subfolder to only get test files at the end.

Reply:
The idea is to not split the test by `.py` files, but instead keep all tests in a model folder, say `tests/models/bert` in one executor, and another one say `tests/models/gpt2` all in (potentially) another executor.